src/main/java/com/shinyhut/vernacular/client/VernacularClient.java

```diff
@@ -47,16 +47,26 @@ public VernacularClient(VernacularConfig config) {
      * @param port Remote port to connect to
      * @throws IllegalStateException if the client is already running
      */
-    public void start(String host, int port) {
+    public void start(String host, int port) throws IOException {
+        startSocket(new Socket(host, port));
```

Comment:
Rather than change this method's signature to throw the IOException, let's preserve the existing behaviour and pass the exception to any registered error listeners, e.g.

```
    public void start(String host, int port) {
        try {
            startSocket(new Socket(host, port));
        } catch (IOException e) {
            handleError(new UnexpectedVncException(e));
        }
    }
```

Reply:
Aha, that's lovely